ssl/ssl_lib.c

```diff
@@ -3889,6 +3892,22 @@ void SSL_set_not_resumable_session_callback(SSL *ssl,
                       (void (*)(void))cb);
 }
 
+void SSL_CTX_set_choose_cipher_callback(SSL_CTX *ctx,
+                                        const SSL_CIPHER* (*cb) (SSL *ssl,
+                                                                 STACK_OF(SSL_CIPHER) *client))
+{
+    SSL_CTX_callback_ctrl(ctx, SSL_CTRL_SET_CHOOSE_CIPHER_CB,
+                          (void (*)(void))cb);
+}
+
+void SSL_set_choose_cipher_callback(SSL *ssl,
+                                    const SSL_CIPHER* (*cb) (SSL *ssl,
+                                                             STACK_OF(SSL_CIPHER) *client))
+{
+    SSL_callback_ctrl(ssl, SSL_CTRL_SET_CHOOSE_CIPHER_CB,
+                      (void (*)(void))cb);
+}
```

Comment:
I am itching to get rid of SSL_ctrl() and avoid it wherever possible. I can't do that though in a binary/source compatible release :-(. Others on the team may have a different view.